packages/wonder-blocks-switch/src/components/switch.tsx

```diff
@@ -0,0 +1,215 @@
+import {StyleSheet} from "aphrodite";
+
+import {
+    UniqueIDProvider,
+    View,
+    addStyle,
+} from "@khanacademy/wonder-blocks-core";
+import Color, {mix} from "@khanacademy/wonder-blocks-color";
+import Icon from "@khanacademy/wonder-blocks-icon";
+
+type Props = {
+    /**
+     * A label for the switch.
+     * If a visible label is not provided, aria-label should be used to
+     * describe the purpose of the switch.
+     */
+    ariaLabel?: string;
+    /**
+     * Whether this compoonent is checked.
+     */
+    checked: boolean;
+    /**
+     * Whether the switch is disabled. Defaults to `false`.
+     */
+    disabled?: boolean;
+    /**
+     * Optional icon to display on the slider.
+     */
+    icon?: React.ReactElement<typeof Icon>;
+    /**
+     * The unique identifier for the switch.
+     */
+    id?: string;
+    /**
+     * Function to call when the switch is clicked.
+     * @param newCheckedValue
+     * @returns {unknown}
+     */
+    onChange?: (newCheckedState: boolean) => unknown;
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string;
+};
+
+const StyledSpan = addStyle("span");
+const StyledInput = addStyle("input");
+
+const Switch = React.forwardRef(function Switch(
+    props: Props,
+    ref: React.ForwardedRef<HTMLInputElement>,
+) {
+    const {
+        ariaLabel,
+        checked,
+        disabled = false,
+        icon,
+        id,
+        onChange,
+        testId,
+    } = props;
+
+    const handleClick = () => {
+        if (!disabled && onChange) {
+            onChange(!checked);
+        }
+    };
+
+    const stateStyles = _generateStyles(checked, disabled);
+
+    let styledIcon: React.ReactElement<typeof Icon> | undefined;
+    if (icon) {
+        styledIcon = React.cloneElement(icon, {
+            size: "small",
+            style: [sharedStyles.icon, stateStyles.icon],
+            ...icon.props,
+        } as Partial<React.ComponentProps<typeof Icon>>);
+    }
+
+    return (
+        <UniqueIDProvider mockOnFirstRender={true} scope="switch">
+            {(ids) => {
+                const uniqueId = id || ids.get("switch");
+
+                return (
+                    <View
+                        onClick={handleClick}
+                        style={[
+                            sharedStyles.switch,
+                            stateStyles.switch,
+                            disabled && sharedStyles.disabled,
+                        ]}
+                        testId={testId}
+                    >
+                        <StyledInput
+                            aria-checked={checked}
+                            aria-disabled={disabled}
+                            aria-label={ariaLabel}
+                            disabled={disabled}
+                            id={uniqueId}
+                            ref={ref}
+                            role="switch"
+                            style={[sharedStyles.hidden]}
```

Comment:
```suggestion
                            style={sharedStyles.hidden}
```

Reply:
Could you also leave a comment pointing out that this input is visually hidden and explain why?